test/remote/gateways/remote_redsys_rest_test.rb

```diff
@@ -0,0 +1,206 @@
+require 'test_helper'
+
+class RemoteRedsysRestTest < Test::Unit::TestCase
+  def setup
+    @gateway = RedsysRestGateway.new(fixtures(:redsys_rest))
+    @amount = 100
+    @credit_card = credit_card('4548812049400004')
+    @declined_card = credit_card
+    @threeds2_credit_card = credit_card('4918019199883839')
+
+    @threeds2_credit_card_frictionless = credit_card('4548814479727229')
+    @threeds2_credit_card_alt = credit_card('4548817212493017')
+    @options = {
+      order_id: generate_order_id
+    }
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'Transaction Approved', response.message
+  end
+
+  def test_purchase_with_invalid_order_id
+    response = @gateway.purchase(@amount, @credit_card, order_id: "a%4#{generate_order_id}")
```

Comment:
Is this invalid order Id test testing that we can clean up the order id as noted in the gateway, or why do we expect success here?

Reply:
Yep! The external contributor added this in their original PR to ensure we clean the order_id